src/components/Api.js

```diff
@@ -0,0 +1,91 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _checkRes(res) {
+    if (!res.ok) {
+      throw new Error("Request failed");
+    }
+    return res.json();
+  }
+
+  _handleError(error) {
+    console.error("API request error: ", error);
+  }
+
+  getUserData() {
+    return fetch(`https://around-api.en.tripleten-services.com/v1/users/me`, {
+      method: "GET",
+      headers: this._headers,
+    }).then(this._checkRes);
+  }
+
+  updateProInfo(name, about) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    }).then(this._checkRes);
```

Comment:
**COULD BE IMPROVED**

You can make a universal method `_request` not to duplicate the `ok` checking, `headers` and `baseUrl`


```js
 _request(endpoint, options = {}) {
    const finalOptions = {
      // add the default headers
      headers: this._headers,
      // spread the other options if provided
      ...options,
    };
      // add the endpoint to the base url
    const url = `${this._baseUrl}${endpoint}`;
    return fetch(url, finalOptions).then(this._checkRes);
  }

  // usage

  getUserInfo() {
    // pass only the endpoint here
    return this._request("/users/me");
  }

  getInitialCards() {
    // pass only the endpoint here
    return this._request("/cards");
  }

  // and so on



```